app/jobs/backorder_job.rb

```diff
@@ -38,35 +38,46 @@ def perform(order)
 
   def place_backorder(order)
     user = order.distributor.owner
-    linked_variants = order.variants.select do |variant|
-      # TODO: scope variants to hub.
-      # We are only supporting producer stock at the moment.
-      variant.on_hand&.negative? && variant.semantic_links.present?
-    end
+    items = backorder_items(order)
 
     # We are assuming that all variants are linked to the same wholesale
     # shop and its catalog:
-    urls = FdcUrlBuilder.new(linked_variants[0].semantic_links[0].semantic_id)
+    reference_link = items[0].variant.semantic_links[0].semantic_id
+    urls = FdcUrlBuilder.new(reference_link)
     orderer = FdcBackorderer.new(user, urls)
 
     backorder = orderer.find_or_build_order(order)
     broker = load_broker(order.distributor.owner, urls)
     ordered_quantities = {}
 
-    linked_variants.each do |variant|
-      retail_quantity = add_item_to_backorder(variant, broker, backorder, orderer)
-      ordered_quantities[variant] = retail_quantity
+    items.each do |item|
+      retail_quantity = add_item_to_backorder(item, broker, backorder, orderer)
+      ordered_quantities[item] = retail_quantity
     end
 
     place_order(user, order, orderer, backorder)
 
-    linked_variants.each do |variant|
-      variant.on_hand += ordered_quantities[variant]
+    items.each do |item|
+      variant = item.variant
+      variant.on_hand += ordered_quantities[item] if variant.on_demand
     end
   end
 
-  def add_item_to_backorder(variant, broker, backorder, orderer)
-    needed_quantity = -1 * variant.on_hand
+  # We look at linked variants which are either stock controlled or
+  # are on demand with negative stock.
+  def backorder_items(order)
```

Comment:
The method sounds like it performs the action (make a backorder), but it just selects the items. Perhaps:
```suggestion
  def backorderable_items(order)
```